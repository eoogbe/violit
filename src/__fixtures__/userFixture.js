/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const username = 'fred-astaire';

export const password = 'password123';

export const credentials = {
  username,
};

export const authUrl = 'http://localhost:4000/api/auth';

export const usersUrl = 'http://localhost:4000/api/users';

export const userUrl = 'http://localhost:4000/api/users/fred-astaire';

export const authApi = [
  {
    href: authUrl,
    data: [
      { name: 'username', value: username },
    ],
  }
];

export const userApi = [
  {
    href: userUrl,
    data: [
      { name: 'username', value: username },
    ],
  }
];

export const credentialsFetch = {
  url: authUrl,
  method: 'GET',
  status: 200,
  body: {
    items: userApi,
  },
};

export const loginFetch = {
  url: authUrl,
  method: 'POST',
  status: 201,
  body: {
    items: userApi,
  },
};

export const logoutFetch = {
  url: authUrl,
  method: 'DELETE',
  status: 200,
};

export const signupFetch = {
  url: usersUrl,
  method: 'POST',
  status: 201,
  body: {
    items: userApi,
  },
};

export const hasUserFetch = {
  url: userUrl,
  method: 'HEAD',
  status: 200,
};

export const doesNotHaveUserFetch = {
  url: userUrl,
  method: 'HEAD',
  status: 404,
};
