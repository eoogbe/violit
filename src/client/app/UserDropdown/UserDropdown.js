/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useRef, useState } from 'react';
import { MdArrowDropDown, MdExitToApp } from 'react-icons/md';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import useOnClickOutside from 'use-onclickoutside';
import { actions as authActions, select as authSelect } from 'client/auth';
import { DropdownMenu } from 'client/dropdown';

const StyledUserDropdown = styled.nav`
  justify-self: end;
  position: relative;
`;

const Toggle = styled.button`
  font-size: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: ${({ theme }) => theme.spacingSm} ${({ theme }) => theme.spacingMd};
  border: 1px solid ${({ theme }) => theme.gray200};
  border-radius: ${({ theme }) => theme.borderRadius};
  background: transparent;
`;

Toggle.displayName = 'Toggle';

/** Dropdown navigation related to the current user. */
const UserDropdown = () => {
  const [open, setOpen] = useState(false);
  const username = useSelector(authSelect.getUsername);
  const dispatch = useDispatch();

  const ref = useRef();
  useOnClickOutside(ref, () => setOpen(false));

  const menuItems = [
    {
      onClick: () => dispatch(authActions.logoutTriggered()),
      icon: MdExitToApp,
      text: 'Log out',
    },
  ];

  return (
    <StyledUserDropdown ref={ref}>
      <Toggle
        type="button"
        aria-haspopup="true"
        aria-expanded={open}
        aria-controls="dropdown-menu-user"
        onClick={() => setOpen(!open)}
      >
        {username}
        <MdArrowDropDown
          size="1.25em"
          attr={{ focusable: 'false', viewBox: '0 0 24 24' }}
        />
      </Toggle>
      {open && <DropdownMenu id="dropdown-menu-user" items={menuItems} />}
    </StyledUserDropdown>
  );
};

export default UserDropdown;
