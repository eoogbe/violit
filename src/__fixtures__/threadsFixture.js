/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as USER from './userFixture';

export const thread1 = {
  id: 'thread1',
  slug: 'lorem-ipsum',
  dateCreated: '2019-09-29T01:23:45.679Z',
  dateDeleted: '2019-09-29T02:00:45.679Z',
  headline: 'Lorem ipsum',
  articleBody: 'Dolor sit amet, consectetur adipiscing elit.',
  author: USER.username,
  title: 'Lorem ipsum | Violit',
  path: '/threads/lorem-ipsum',
  url: 'http://localhost:4000/api/threads/lorem-ipsum',
  deleteUrl: 'http://localhost:4000/api/threads/lorem-ipsum/delete',
  commentsUrl: 'http://localhost:4000/api/threads/lorem-ipsum/comments',
};

export const thread2 = {
  id: 'thread2',
  slug: 'aliquip-ex-ea',
  dateCreated: '2019-09-28T09:08:07.654Z',
  dateDeleted: '2019-09-28T09:08:08.654Z',
  headline: 'Aliquip ex ea',
  articleBody: null,
  author: USER.username,
  title: 'Aliquip ex ea | Violit',
  path: '/threads/aliquip-ex-ea',
  url: 'http://localhost:4000/api/threads/aliquip-ex-ea',
  deleteUrl: 'http://localhost:4000/api/threads/aliquip-ex-ea/delete',
  commentsUrl: 'http://localhost:4000/api/threads/aliquip-ex-ea/comments',
};

export const thread3 = {
  id: 'thread3',
  slug: 'commodo-consequat',
  dateCreated: '2019-09-27T03:21:09.876Z',
  dateDeleted: '2019-09-27T04:21:09.876Z',
  headline: 'Commodo consequat',
  articleBody: null,
  author: USER.username,
  title: 'Commodo consequat | Violit',
  path: '/threads/commodo-consequat',
  url: 'http://localhost:4000/api/threads/commodo-consequat',
  deleteUrl: 'http://localhost:4000/api/threads/commodo-consequat/delete',
  commentsUrl: 'http://localhost:4000/api/threads/commodo-consequat/comments',
};

export const createInput = {
  headline: thread1.headline,
  articleBody: thread1.articleBody,
};

export const normalized = {
  thread1: {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    dateDeleted: null,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    author: thread1.author,
  },
  thread2: {
    id: 'thread2',
    slug: thread2.slug,
    dateCreated: thread2.dateCreated,
    dateDeleted: thread2.dateDeleted,
    headline: thread2.headline,
    articleBody: thread2.articleBody,
    author: thread2.author,
  },
  thread3: {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    dateDeleted: null,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    author: thread3.author,
  },
};

export const unnormalized = [
  {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    author: thread1.author,
  },
  {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    author: thread3.author,
  },
];

export const api = [
  {
    href: thread1.url,
    data: [
      { name: 'id', value: 'thread1' },
      { name: 'slug', value: thread1.slug },
      { name: 'dateCreated', value: thread1.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'headline', value: thread1.headline },
      { name: 'articleBody', value: thread1.articleBody },
      { name: 'author', value: thread1.author },
    ],
    links: [
      { rel: 'deleteAction', href: thread1.deleteUrl },
      { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
      {
        rel: 'comment',
        href: thread1.commentsUrl,
      },
    ],
  },
  {
    href: thread3.url,
    data: [
      { name: 'id', value: 'thread3' },
      { name: 'slug', value: thread3.slug },
      { name: 'dateCreated', value: thread3.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'headline', value: thread3.headline },
      { name: 'articleBody', value: '' },
      { name: 'author', value: thread3.author },
    ],
    links: [
      { rel: 'deleteAction', href: thread3.deleteUrl },
      { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
      {
        rel: 'comment',
        href: thread3.commentsUrl,
      },
    ],
  },
];

export const deleteThreadApi = [
  {
    href: thread1.deleteUrl,
    data: [
      { name: 'actionStatus', value: 'completedActionStatus' },
      { name: 'endTime', value: thread1.dateDeleted },
    ],
    links: [
      { rel: 'object', href: thread1.url },
      { rel: 'targetCollection', href: 'http://localhost:4000/api/board' },
    ],
  },
];

export const threadsFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'GET',
  status: 200,
  body: {
    items: api,
  },
};

export const threadFetch1 = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [api[0]],
  },
};

export const threadWithoutArticleBodyFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: '' },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: '' },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'deleteAction', href: thread1.deleteUrl },
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const threadWhereCannotCreateCommentFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: '' },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: thread1.articleBody },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const deletedThreadFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: thread1.dateDeleted },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: '' },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const createThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'POST',
  status: 201,
  body: {
    items: [api[0]],
  },
};

export const canCreateThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS,POST',
  },
};

export const cannotCreateThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS',
  },
};

export const deleteThreadFetch = {
  url: thread1.deleteUrl,
  method: 'POST',
  status: 200,
  body: {
    items: deleteThreadApi,
  },
};

export const db = [
  {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    dateDeleted: null,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
  {
    id: 'thread2',
    slug: thread2.slug,
    dateCreated: thread2.dateCreated,
    dateDeleted: thread2.dateDeleted,
    headline: thread2.headline,
    articleBody: thread2.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
  {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    dateDeleted: null,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
];
