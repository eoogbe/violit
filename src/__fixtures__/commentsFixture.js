/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as THREADS from './threadsFixture';
import * as USER from './userFixture';

export const comment1 = {
  id: 'comment1',
  dateCreated: '2019-10-09T10:11:12.123Z',
  dateDeleted: '2019-10-09T12:11:12.123Z',
  text: 'Sed do eiusmod tempor incididunt ut labore et dolore.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const comment2 = {
  id: 'comment2',
  dateCreated: '2019-10-08T07:08:09.101Z',
  dateDeleted: '2019-10-08T07:09:09.101Z',
  text: 'Ut enim ad minim veniam.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const comment3 = {
  id: 'comment3',
  dateCreated: '2019-09-30T01:02:03.456Z',
  dateDeleted: '2019-10-03T01:02:03.456Z',
  text: 'Quis nostrud exercitation ullamco laboris nisi ut.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const createInput = {
  id: 'comment1',
  text: comment1.text,
  threadSlug: THREADS.thread1.slug,
};

export const normalized = {
  comment1: {
    id: 'comment1',
    dateCreated: comment1.dateCreated,
    dateDeleted: null,
    text: comment1.text,
    threadSlug: comment1.threadSlug,
    author: comment1.author,
  },
  comment2: {
    id: 'comment2',
    dateCreated: comment2.dateCreated,
    dateDeleted: comment2.dateDeleted,
    text: comment2.text,
    threadSlug: comment2.threadSlug,
    author: comment2.author,
  },
  comment3: {
    id: 'comment3',
    dateCreated: comment3.dateCreated,
    dateDeleted: null,
    text: comment3.text,
    threadSlug: comment3.threadSlug,
    author: comment3.author,
  },
};

export const unnormalized = [
  normalized.comment1,
  normalized.comment2,
  normalized.comment3,
];

export const api = [
  {
    href: 'http://localhost:4000/api/comments/comment1',
    data: [
      { name: 'id', value: 'comment1' },
      { name: 'dateCreated', value: comment1.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'text', value: comment1.text },
      { name: 'author', value: comment1.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      {
        rel: 'deleteAction',
        href: 'http://localhost:4000/api/comments/comment1/delete',
      },
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
  {
    href: 'http://localhost:4000/api/comments/comment2',
    data: [
      { name: 'id', value: 'comment2' },
      { name: 'dateCreated', value: comment2.dateCreated },
      { name: 'dateDeleted', value: comment2.dateDeleted },
      { name: 'text', value: comment2.text },
      { name: 'author', value: comment2.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
  {
    href: 'http://localhost:4000/api/comments/comment3',
    data: [
      { name: 'id', value: 'comment3' },
      { name: 'dateCreated', value: comment3.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'text', value: comment3.text },
      { name: 'author', value: comment3.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      {
        rel: 'deleteAction',
        href: 'http://localhost:4000/api/comments/comment3/delete',
      },
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
];

export const deleteCommentApi = [
  {
    href: 'http://localhost:4000/api/comments/comment1/delete',
    data: [
      { name: 'actionStatus', value: 'completedActionStatus' },
      { name: 'endTime', value: comment1.dateDeleted },
    ],
    links: [
      {
        rel: 'object',
        href: 'http://localhost:4000/api/comments/comment1',
      },
      { rel: 'targetCollection', href: THREADS.thread1.commentsUrl },
    ],
  },
];

export const commentsFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'GET',
  status: 200,
  body: {
    items: api,
  },
};

export const createCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'POST',
  status: 201,
  body: {
    items: [api[0]],
  },
};

export const canCreateCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS,POST',
  },
};

export const cannotCreateCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS',
  },
};

export const deleteCommentFetch = {
  url: 'http://localhost:4000/api/comments/comment1/delete',
  method: 'POST',
  status: 200,
  body: {
    items: deleteCommentApi,
  },
};

export const db = [
  {
    id: 'comment1',
    dateCreated: comment1.dateCreated,
    dateDeleted: null,
    content: comment1.text,
    userId: 'user1',
    threadId: 'thread1',
  },
  {
    id: 'comment2',
    dateCreated: comment2.dateCreated,
    dateDeleted: comment2.dateDeleted,
    content: comment2.text,
    userId: 'user1',
    threadId: 'thread1',
  },
  {
    id: 'comment3',
    dateCreated: comment3.dateCreated,
    dateDeleted: null,
    content: comment3.text,
    userId: 'user1',
    threadId: 'thread1',
  },
];
