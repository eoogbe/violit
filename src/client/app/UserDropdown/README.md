```jsx
import * as USER from '__fixtures__/userFixture';
import { constants as authConstants } from 'client/auth';

const state = {
  [authConstants.NAME]: {
    credentials: USER.credentials,
  },
};

<StoreProvider state={state}>
  <div style={{ marginLeft: '50px', width: '150px' }}>
    <UserDropdown />
  </div>
</StoreProvider>
```
