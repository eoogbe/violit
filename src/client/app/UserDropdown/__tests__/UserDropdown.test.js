/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import * as R from 'ramda';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import * as USER from '__fixtures__/userFixture';
import {
  actions as authActions,
  constants as authConstants,
} from 'client/auth';
import UserDropdown from '../UserDropdown';

const STATE = {
  [authConstants.NAME]: {
    credentials: USER.credentials,
  },
};

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore(STATE);
  const renderer = TestRenderer.create(
    <Provider store={store}>
      <UserDropdown />
    </Provider>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it('renders the open menu', () => {
  const store = mockStore(STATE);
  const renderer = TestRenderer.create(
    <Provider store={store}>
      <UserDropdown />
    </Provider>
  );
  const instance = renderer.root;

  const toggle = instance.find(R.pathEq(['type', 'displayName'], 'Toggle'));
  TestRenderer.act(() => {
    toggle.props.onClick();
  });
  expect(renderer.toJSON()).toMatchSnapshot();
});

it('triggers logout', () => {
  const store = mockStore(STATE);
  const wrapper = mount(
    <Provider store={store}>
      <UserDropdown />
    </Provider>
  );

  wrapper.find('Toggle').simulate('click');
  wrapper.find('MenuItemButton').first().simulate('click');

  expect(store.getActions()).toEqual([authActions.logoutTriggered()]);
});
