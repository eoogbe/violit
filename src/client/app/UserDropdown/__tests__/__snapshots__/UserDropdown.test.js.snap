// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the open menu 1`] = `
<nav
  className="sc-chPdSV joRJWR"
>
  <button
    aria-controls="dropdown-menu-user"
    aria-expanded={true}
    aria-haspopup="true"
    className="sc-kgoBCf cQygcs"
    onClick={[Function]}
    type="button"
  >
    fred-astaire
    <svg
      fill="currentColor"
      focusable="false"
      height="1.25em"
      size="1.25em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1.25em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </button>
  <ul
    className="sc-cSHVUG eBHCFa"
    id="dropdown-menu-user"
    role="menu"
  >
    <li
      role="menuitem"
    >
      <button
        className="sc-kAzzGY cbGqLE"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1.25em"
          size="1.25em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1.25em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        Log out
      </button>
    </li>
  </ul>
</nav>
`;

exports[`renders without crashing 1`] = `
<nav
  className="sc-chPdSV joRJWR"
>
  <button
    aria-controls="dropdown-menu-user"
    aria-expanded={false}
    aria-haspopup="true"
    className="sc-kgoBCf cQygcs"
    onClick={[Function]}
    type="button"
  >
    fred-astaire
    <svg
      fill="currentColor"
      focusable="false"
      height="1.25em"
      size="1.25em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1.25em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </button>
</nav>
`;
